IRGen: adjust for LLVM SVN r368000

compnerd opened the pull request:
<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves SR-NNNN.

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

compnerd commented:
CC: @rjmccall

compnerd commented:
@swift-ci please smoke test

rjmccall commented on lib/IRGen/GenDecl.cpp:
```diff
@@ -4163,7 +4163,8 @@ Address IRGenFunction::createAlloca(llvm::Type *type,
       new llvm::AllocaInst(type, IGM.DataLayout.getAllocaAddrSpace(), ArraySize,
-                           alignment.getValue(), name, AllocaIP);
+                           llvm::MaybeAlign(alignment.getValue()), name,
+                           AllocaIP);
```
Do we ever actually use `Alignment(0)` as input to this?

compnerd replied:
I'm not certain, but I do agree with you that we shouldn't be passing 0 - the signature for `AllocaInst` has changed and now requires the `MaybeAlign` type (as a wrapper to the alignment).

rjmccall replied:
Right, I'm just wondering if maybe we can pass `llvm::Align(alignment.getValue())` (or even have an implicit conversion, and maybe someday unify the types).

compnerd committed:
IRGen: use `llvm::Align` rather than `llvm::MaybeAlign`

The alignment should never be 0, use the explicit `llvm::Align` rather
than `llvm::MaybeAlign`.

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test
